### PR TITLE
Shadow documentation overhaul

### PR DESCRIPTION
**Pull request by jongerrish**

### CODE REVIEW

File: extending.md [hunk 1]
```diff
@@ -20,8 +23,6 @@ Why "Shadow?" Shadow objects are not quite [Proxies](http://en.wikipedia.org/wik
...
 
-From Robolectric 2.0 on, the number of shadow classes needed is greatly reduced, because real Android OS code is present. Methods on your shadow class are able to call through to the Android OS code if you like, using <code>Robolectric.directlyOn()</code>.
-
 ### Shadow Classes
 
 Shadow classes always need a public no-arg constructor so that the Robolectric framework can instantiate them. They are associated to the class that they Shadow with an `@Implements` annotation on the class declaration. In general, they should be implemented as if from scratch, the facilities of the classes they Shadow have almost always been removed and their data members are difficult to access. The methods on a Shadow class usually either Shadow the methods on the original class or facilitate testing by setting up return values or providing access to internal state or logged method calls.
```
Comment (xian) [inline extending.md]: > In general, they should be implemented as if from scratch, the facilities of the classes they Shadow have almost always been removed and their data members are difficult to access. The methods on a Shadow class usually either Shadow the methods on the original class or facilitate testing by setting up return values or providing access to internal state or logged method calls.

This graf doesn't make much sense to me, maybe just kill it?

File: extending.md [hunk 2]
```diff
@@ -55,17 +56,21 @@ Shadow methods must be marked with the `@Implementation` annotation. Robolectric
...
   ...	
   @Implementation
-  public void setImageResource(int resId) {
+  protected void setImageResource(int resId) {
     // implementation here.
   }
 }
 ```
 
+Robolectric supports shadowing all methods on the original class, including private, static, final or native.
```
Comment (xian) [inline extending.md]: Convention is to wrap code/keywords/symbol names in backticks.

File: extending.md [hunk 3]
```diff
@@ -79,6 +84,7 @@ Robolectric would invoke the following  `__constructor__` method that receives a
 @Implements(TextView.class)
 public class ShadowTextView {
   ...
+  @Implementation
   public void __constructor__(Context context) {
```
Comment (xian) [inline extending.md]: This can be `protected` now.

File: extending.md [hunk 4]
```diff
@@ -104,3 +110,8 @@ public class ShadowPoint {
 Robolectric will set realPoint to the actual instance of `Point` before invoking any other methods.
 
 It is important to note that methods called on the real object will still be intercepted and redirected by Robolectric. This does not often matter in test code, but it has important implications for Shadow class implementors. Since the Shadow class inheritance hierarchy does not always mirror that of their associated Android classes, it is sometimes necessary to make calls through these real objects so that the Robolectric runtime will have the opportunity to route them to the correct Shadow class based on the actual class of the object. Otherwise methods on Shadows of base classes would be unable to access methods on the Shadows of their subclasses.
+
+Methods on your shadow class are able to call through to the Android OS code, using <code>Shadow.directlyOn()</code>.
+
+
+Robolectric is a work in progress, and we welcome contributions from the community. We encourage developers to [use the standard GitHub workflow](http://help.github.com/fork-a-repo/ "Help.GitHub - Fork A Repo") to fork, enhance, and submit pull requests to us.
```
Comment (xian) [inline extending.md]: Maybe add a pointer to [Contributing](http://robolectric.org/contributing/)?

File: extending.md (hunk 1 above)
Comment (xian) [reply extending.md]: I think it's a hold-over from 1.0 maybe.

Comment (jongerrish) [reply extending.md]: Removed.

File: extending.md (hunk 2 above)
Comment (jongerrish) [reply extending.md]: Done

**Pull request merged**